src/core/renderer.js

```diff
@@ -179,24 +203,25 @@ class Renderer {
                     continue;
                 }
 
-                const feature = this._featureFromDataFrame(dataframe, i);
-
                 // Ignore filtered features
                 if (viz.filter.eval(feature) < FILTERING_THRESHOLD) {
                     continue;
                 }
 
-                viewportExpressions.forEach(viewportExpression => viewportExpression.accumViewportAgg(feature));
+                for (let j = 0; j < viewportExpressionsLength; j++) {
```

Comment:
what was the x% improvement of using a for loop?

Reply:
I didn't check it with just that change, sorry, anyway, the for loop is quite hot